samples/led-matrix-weather/Program.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Iot.Device.LEDMatrix;
+using Iot.Device.Graphics;
+using System.Drawing;
+using System.Numerics;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Sockets;
+using System.Net.NetworkInformation;
+
+namespace LedMatrixWeather
+{
+    partial class Program
+    {
+        static Action<RGBLedMatrix> s_scenario = WeatherDemo;
+        static Stopwatch s_showLocalIp = null;
+        static string[] s_ips;
+        static bool s_networkAvailable = false;
+        static Weather s_client;
+        static OpenWeatherResponse s_weatherResponse;
+        static TimeZoneInfo s_timeZonePst = TimeZoneInfo.FindSystemTimeZoneById("America/Los_Angeles");
+
+        static void Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Console.WriteLine("Usage: led-matrix-weather <openweathermap.org API key>");
+                Console.WriteLine();
+                Console.WriteLine("If wrong key is used or there is no network connection");
+                Console.WriteLine("example data will be displayed.");
+                Console.WriteLine();
+                Console.WriteLine("For the first 30 seconds of execution");
+                Console.WriteLine("IP addresses will be displayed instead of data.");
+                return;
+            }
+
+            s_client = new Weather(args[0]);

```

Comment:
NIT: Should we have this inside a try and fail  if the key was invalid?

Reply:
the constructor doesn't validate the key so doesn't matter